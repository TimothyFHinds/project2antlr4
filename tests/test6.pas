program test6;

var 
a, b, c, min: REAL;

//User-defined procedure
//https://www.tutorialspoint.com/pascal/pascal_procedures.htm

procedure findMin(x, y, z: real; var m: real);
    //find the minimum of the three input values
begin
    if(x < y) then
        m := x
        else 
        m := y;
    
    if(z<m) then
        m := z;
end;
//end of findMin

//main block
begin
    writeln('Enter three numbers: ');
    readln(a);
    readln(b);
    readln(c);
    findMin(a, b, c, min);

    writeln('Minimum of your 3 numbers: ');
    writeln(min);
end.