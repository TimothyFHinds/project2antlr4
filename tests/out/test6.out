'Checking for minimum of 3 numbers'
'Minimum of the 3 numbers: '
300.0
